src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -134,11 +137,14 @@ public static class TagHelperBlockRewriter
                     SourceLocation symbolStartLocation;
 
                     // Check for attribute start values, aka single or double quote
-                    if (IsQuote(htmlSymbols[i + 1]))
+                    if (i + 1 < htmlSymbols.Length && IsQuote(htmlSymbols[i + 1]))
```

Comment:
Right now Razor doesn't understand attributes with space around the equals as mentioned on lines 131/132 in this method.

1. `checked` will hit the second branch where we find the name. We will also do work to understand that we didn't find a value.
1. `@multiple` wont be in this method since that will be wrapped in a Block and will be handled by the `TryParseBlock` method.
1. `type = @type` will hit the other method `TryParseBlock` because it's understood as having a "dynamic" value.
1. type = "@type" same as above